Purchases/Caching/DeviceCache.swift

```diff
@@ -493,4 +499,38 @@ fileprivate extension UserDefaults {
 
 }
 
+private extension DeviceCache {
+    enum CacheDuration {
+        // swiftlint:disable:next nesting
+        enum AppStatus {
+            case foreground
+            case background
+
+            init(backgrounded: Bool) {
+                self = backgrounded ? .background : .foreground
+            }
+        }
+
+        // swiftlint:disable:next nesting
+        enum Environment {
+            case production
+            case sandbox
+
+            init(sandbox: Bool) {
+                self = sandbox ? .sandbox : .production
+            }
+        }
+
+        static func duration(status: AppStatus, environment: Environment) -> TimeInterval {
```

Comment:
Welcoming feedback here. I think the `enum`s make this a bit clearer than a couple of `Bool`s.

Reply:
YES, this is 💯 I'm a big fan of using `enum`s for this sort of configuration.